libs/hdf-converters/test/attestations/attestations.spec.ts

```diff
@@ -6,108 +6,371 @@ import {
+const aPass_cPass_unexp: Attestation[] = [
+  {
+    control_id: 'SV-230221',
+    explanation: 'This is a test explanation for 230328',
+    frequency: 'monthly',
+    status: ExecJSON.ControlResultStatus.Passed,
+    updated: '3000-01-01',
+    updated_by: 'Attestation Tester'
+  }
+];
+
+const aPass_cPass_exp: Attestation[] = [
+  {
+    control_id: 'SV-230221',
+    explanation: 'This is a test explanation for 230328',
+    frequency: 'monthly',
+    status: ExecJSON.ControlResultStatus.Passed,
+    updated: '1999-01-01',
+    updated_by: 'Attestation Tester'
+  }
+];
+
+const aFail_cPass_unexp: Attestation[] = [
+  {
+    control_id: 'SV-230221',
+    explanation: 'This is a test explanation for 230328',
+    frequency: 'monthly',
+    status: ExecJSON.ControlResultStatus.Failed,
+    updated: '3000-01-01',
+    updated_by: 'Attestation Tester'
+  }
+];
+
+const aFail_cPass_exp: Attestation[] = [
+  {
+    control_id: 'SV-230221',
+    explanation: 'This is a test explanation for 230328',
+    frequency: 'monthly',
+    status: ExecJSON.ControlResultStatus.Passed,
+    updated: '1999-01-01',
+    updated_by: 'Attestation Tester'
+  }
+];
+
+const aPass_cFail_unexp: Attestation[] = [
+  {
+    control_id: 'SV-230222',
+    explanation: 'This is a test explanation for 230222',
     frequency: '1d',
     status: ExecJSON.ControlResultStatus.Passed,
-    updated: '2022-01-02',
-    updated_by: 'Alec Hardison, Security'
+    updated: '3000-01-01',
+    updated_by: 'Attestation Tester'
   }
 ];
 
-describe('attestations', () => {
-  it('Should create a message for attestations, including expiration status if expired', () => {
+describe('advanceDate', () => {
+  it('Should return a date two weeks from now when given "fortnightly" as an input', () => {
+    expect(
+      advanceDate(moment.utc(1662758942000), 'fortnightly').toISOString(true)
+    ).toEqual('2022-09-23T21:29:02.000+00:00');
+  });
+
+  it('Should return correct date when given custom number of days to advance by', () => {
+    expect(
+      advanceDate(moment.utc(1662758942000), '200d').toISOString(true)
+    ).toEqual('2023-03-28T21:29:02.000+00:00');
+  });
+
+  it('Should return correct date when given custom number of weeks to advance by', () => {
+    expect(
+      advanceDate(moment.utc(1662758942000), '12w').toISOString(true)
+    ).toEqual('2022-12-02T21:29:02.000+00:00');
+  });
+
+  it('Should return correct date when given custom number of months to advance by', () => {
+    expect(
+      advanceDate(moment.utc(1662758942000), '4m').toISOString(true)
+    ).toEqual('2023-01-09T21:29:02.000+00:00');
+  });
+
+  it('Should return correct date when given custom number of years to advance by', () => {
+    expect(
+      advanceDate(moment.utc(1662758942000), '5y').toISOString(true)
+    ).toEqual('2027-09-09T21:29:02.000+00:00');
+  });
+});
+
+describe('CreateAttestationMessage', () => {
+  it('Should create a message for an unexpired attestation', () => {
```

Comment:
What do you mean by message? Is this a field that is added to the data? Where does it show up? What does it represent? I would add a little more clarity here.

Reply:
Would it be be more clear if I re-arrange the order of the tests being conducted @em-c-rod? Seeing what addAttestationToHDF() does and the comment I add that describes the HDF json structure for controls and attestations might make the rest more clear if that's read first